pom.xml

```diff
@@ -0,0 +1,553 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.apache.phoenix</groupId>
+    <artifactId>phoenix-queryserver</artifactId>
+    <version>4.15.0-HBase-1.4-SNAPSHOT</version>
```

Comment:
Since this is now in its own repo we should use 1.0.0

Reply:
Phoenix-connectors started off in this way, so I followed it. But yes, we can update it. We will need to have 4 branches here as well. 1.0.0-HBase-1.4, 1.0.0-HBase-1.3, 1.0.0-HBase-1.2 and master

FYI: https://github.com/apache/phoenix-connectors/blob/4.x-HBase-1.4/presto-phoenix-shaded/pom.xml

Reply:
> We will need to have 4 branches here as well

Do we need to have four branches? If we're going to make such a change like this already, I'd love to move away from such a burden.

Reply:
Yeah, we may not need that. I think we can put a default version of phoenix and override others at runtime for those branches. I believe such a behavior is possible with maven and builds can be setup accordingly.

Reply:
Yah, as long as we're "reasonable" with Phoenix API, we can hopefully avoid all that headache :)

Reply:
Fine by me.

Reply:
Here's what I am saying.

This project will be versioned from `1.0.0-SNAPSHOT`.
The project will have a phoenix.version property which will define the versions of phoenix it will use (phoenix-core)
This value will be defaulted to `4.14.1-HBase-1.4` (Doing so will cause compilation failure since we have a patch in 4.15 (PHOENIX-4755) here, so we need to discuss this)
The compilation can be carried out with `mvn clean install -DskipTests -Dphoenix.version=4.15.0-HBase-1.4-SNAPSHOT` or any other version as deemed appropriate.